src/home/homework-luks.c

```diff
@@ -2724,6 +2724,40 @@ static int prepare_resize_partition(
         return 1;
 }
 
+static int get_maximum_partition_size(
+                int fd,
+                struct fdisk_partition *p,
+                uint64_t *ret_maximum_partition_size) {
+
+        _cleanup_(fdisk_unref_contextp) struct fdisk_context *c = NULL;
+        uint64_t start_lba, start, last_lba, end;
+        int r;
+
+        assert(fd >= 0);
+        assert(p);
```

Comment:
so this is actually not correct to assume. if a naked block device is used (i.e. no partition table) this will be NULL.

Reply:
Added `partition != NULL` condition. This will make it the same as before if it's a naked block device.